httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java

```diff
@@ -967,7 +967,7 @@ public List<NameValuePair> getQueryParams() {
      * @since 5.2
      */
     public NameValuePair getFirstQueryParam(final String name) {
-        return queryParams.stream().filter(e -> name.equals(e.getName())).findFirst().orElse(null);
```

Comment:
@skssxf Could you please do a null check here instead of creating an empty list here?